File: BGC_WebApi/BGC_DataAccess/Entities/RpgCampaign.cs
using Microsoft.EntityFrameworkCore.Metadata.Builders;
using Microsoft.EntityFrameworkCore;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BGC_DataAccess.Entities;

public class RpgCampaign
{
    public int RpgCampaignId { get; set; }
    public string Name { get; set; }
    public string? Description { get; set; }

    public RpgCampaignRegistration RpgCampaignRegistration { get; set; }    

    public int? OrganizerId { get; set; }   
    public Member Organizer { get; set; }

    public ICollection<RpgSession> Sessions { get; set; }
}

public class RpgCampaignConfiguration : IEntityTypeConfiguration<RpgCampaign>
{
    public void Configure(EntityTypeBuilder<RpgCampaign> builder)
    {
        builder.Property(r => r.Name).IsRequired().HasMaxLength(50);
        builder.Property(r => r.Description).HasMaxLength(1000);
        builder.Property(r => r.Organizer).IsRequired();
    }
}

File: BGC_Portal/src/app/api/models/rpg-campaign.ts
/* tslint:disable */
/* eslint-disable */
import { Member } from './member';
import { RpgCampaignRegistration } from './rpg-campaign-registration';
import { RpgSession } from './rpg-session';
export interface RpgCampaign {
  description?: null | string;
  name?: null | string;
  organizer?: Member;
  organizerId?: null | number;
  rpgCampaignId?: number;
  rpgCampaignRegistrations?: null | Array<RpgCampaignRegistration>;
  sessions?: null | Array<RpgSession>;
}
